pages/admin/view-templates.js

```diff
@@ -2,26 +2,29 @@
 import DataView from "../../components/containers/Dashboard/DataView";
 import { serverSideTranslations } from "next-i18next/serverSideTranslations";
 import { invokeLambda } from "../api/templates";
+import { requireAuthentication } from "../../lib/auth";
 
-export async function getServerSideProps(context) {
-  // getStaticProps is serverside, and therefore instead of doing a request,
-  // we import the invoke Lambda function directly
-  const payload = {
-    method: "GET",
-  };
-  const lambdaResult = JSON.parse(await invokeLambda(payload));
-  console.log(lambdaResult);
-  const templatesJSON =
-    lambdaResult.data && lambdaResult.data.records && lambdaResult.data.records.length > 0
-      ? lambdaResult.data.records
-      : [];
+export const getServerSideProps = requireAuthentication(async (context) => {
+  {
+    // getStaticProps is serverside, and therefore instead of doing a request,
+    // we import the invoke Lambda function directly
+    const payload = {
+      method: "GET",
+    };
+    const lambdaResult = JSON.parse(await invokeLambda(payload));
+    console.log(lambdaResult);
```

Comment:
Not urgent, nor needed for this PR, just wondering if there's a need for error handling for lambda?

Reply:
Yes, there definitely is. Right now if the lambda fails for whatever reason the whole app crashes.

Reply:
But it looks like the submission lambda is set up the same way? So I'm not sure the urgency